src/main/java/io/pravega/connectors/flink/FlinkPravegaInputFormat.java

```diff
@@ -134,11 +137,14 @@ public BaseStatistics getStatistics(BaseStatistics cachedStatistics) throws IOEx
             BatchClient batchClient = clientFactory.createBatchClient();
 
             for (String stream : streamNames) {
-                for (Iterator<SegmentInfo> segmentInfos = batchClient.listSegments(new StreamImpl(scopeName, stream)); segmentInfos.hasNext(); ) {
-                    SegmentInfo segmentInfo = segmentInfos.next();
-                    Segment segment = segmentInfo.getSegment();
-                    // segmentInfo.getWriteOffset() will give the length of the segment.
-                    splits.add(new PravegaInputSplit(splits.size(), segment, segmentInfo.getStartingOffset(), segmentInfo.getWriteOffset()));
+
+                Iterator<SegmentRange> segmentRangeIterator =
+                        batchClient.getSegments(Stream.of(scopeName, stream), null, null).getIterator();
+                while (segmentRangeIterator.hasNext()) {
+                    SegmentRange segmentRange = segmentRangeIterator.next();
+                    Segment segment = Segment.fromScopedName(segmentRange.getScopeName() + "/"
+                            + segmentRange.getStreamName() + "/" + segmentRange.getSegmentNumber());
```

Comment:
Seems there's a couple of Pravega issues to be reported here:
1. `SegmentRangeImpl` is not serializable and thus cannot be stored in the `PravegaInputSplit`.
2. `SegmentRange.getSegment()` returns a corrupted `Segment` object (the scope is wrong).  This forces us to reconstruct the segment object.

Reply:
For (1), I have created a JIRA ticket https://github.com/pravega/pravega/issues/2433

For (2), I don't see an API to get the segment reference from segment range directly `SegmentRange.getSegment()`. 

I was trying to create a new segment instance (https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/segment/impl/Segment.java#L38) which apparently was not working (corrupted segment). The issue got resolved when I try to create the segment using `Segment.fromScopedName()` API.

Reply:
(1) I think this PR solves the issue, yes? https://github.com/pravega/pravega/pull/2434

(2) I don't see any specific reason for requiring  a `Segment` object in `PravegaInputStream`. Perhaps we should simply replace segment `Segment` with  `SegmentRange` in the constructor and internally. Would it work?

Reply:
On (2), yes I was using Segment instance because of (1). Now that (1) is resolved, I will pull the change and modify `PravegaInputStream` to use only `SegmentRange`